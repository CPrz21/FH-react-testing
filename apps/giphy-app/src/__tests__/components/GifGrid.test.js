import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifGrid} from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('<GifGrid />', () => {
  
  test('should show component correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category="joke" />);
    expect(wrapper).toMatchSnapshot();
  })

  test('should show items when images are loaded from useFetchGifs', () => {
    const gifs = [{
        id:'ABC',
        url: 'https://localhost/any/thing.jpg',
        title: 'Invisible Image'
      },
      {
        id:'CBA',
        url: 'https://localhost/any/thing.jpg',
        title: 'Invisible Image'
      }
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow( < GifGrid category="Jest" /> );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })
  
  
})
