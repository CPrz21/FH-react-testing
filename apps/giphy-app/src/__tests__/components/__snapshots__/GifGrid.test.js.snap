// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> should show component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    joke
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> should show items when images are loaded from useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Jest
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Invisible Image"
      url="https://localhost/any/thing.jpg"
    />
  </div>
</Fragment>
`;
